src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -1094,6 +1094,70 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
                 operandStack.push_back(builder.CreateLoad(builder.getInt32Ty(), locals[3]));
                 break;
             }
+            case OpCodes::IfEq:
+            {
+                auto target = consume<std::int16_t>(current);
+                llvm::BasicBlock* basicBlock = basicBlocks[target + offset];
+                llvm::BasicBlock* next = basicBlocks[current.data() - code.getCode().data()];
+
+                llvm::Value* val = operandStack.pop_back(builder.getInt32Ty());
+                llvm::Value* zero = builder.getInt32(0);
+
+                llvm::Value* cond = builder.CreateICmpEQ(val, zero);
```

Comment:
There's a lot of code duplication in these methods where the only difference between e.g. `IfEq` and `IfGe` is the call to `CreateICmpEQ`. 
Could you instead make one single code for these and then within the implementation use `builder.CreateICmp` with the appropiate enum value to avoid the duplication?

So roughly:
```cpp
case IfEq:
case IfGe:
case ...:
{
    ...
    switch(opCode)
    {
       case IfEq: predicate = ICMP_EQ; break;
       case IfGe: predicate = ICMP_SGE; break;
       ...
    }
    builder.CreateICmp(predicate, ...);
    ...
}
```

ditto for the `If<cmp>` versions